core/src/main/java/org/apache/brooklyn/enricher/stock/Enrichers.java

```diff
@@ -171,6 +173,9 @@ public PropagatorBuilder propagatingAllBut(Iterable<? extends Sensor<?>> vals) {
         public JoinerBuilder joining(AttributeSensor<?> source) {
             return new JoinerBuilder(source);
         }
+        public <S, T> ReducerBuilder<S, T> reducing(Class<? extends Reducer<S, T>> clazz, List<AttributeSensor<S>> sourceSensors) {
```

Comment:
Would be nice to have an additional method factory which just uses the generic `Reducer` implementation, working with `Function<List<?>, ?>` params - which I guess would be the main way it's used in java land.